src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+  const dateObject = {};
+  const newDate = [];
```

Comment:
Consider renaming 'newDate' to 'formattedDateComponents' to better represent its purpose.